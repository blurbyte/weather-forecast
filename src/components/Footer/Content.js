import styled from 'styled-components';

import BaseContent from '../Content';

const Content = styled(BaseContent)`
  display: flex;
  align-items: center;
  height: 100%;
`;

export default Content;
