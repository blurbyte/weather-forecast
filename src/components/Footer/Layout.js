import styled from 'styled-components';
import { transparentize } from 'polished';

const Wrapper = styled.footer`
  height: 6rem;
  border-top: 1px solid ${props => transparentize(0.5, props.theme.primary)};
  grid-area: footer;
`;

export default Wrapper;
