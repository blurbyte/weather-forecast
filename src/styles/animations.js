import { keyframes } from 'styled-components';

export const fadeIn = keyframes`
  from {
    opacity: 0;
    transform: translateY(2rem);
  }
  to {
    opacity: 1;
    transform: translateY(0);
  }
`;
