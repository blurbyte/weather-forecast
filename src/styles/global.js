import { injectGlobal } from 'styled-components';

injectGlobal`
  @import url('https://fonts.googleapis.com/css?family=Cardo:700&subset=latin-ext');

  html {
    font-size: 62.5%;
  }

  body {
    background: #71d7ed;
    color: #222222;
    font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif;
    line-height: 1.4;
    font-size: 1.4rem;
    font-weight: 400;
    overflow-y: scroll;
    user-select: none;
  }
`;
