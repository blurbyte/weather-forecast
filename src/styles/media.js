// Generates media templates based on device breakpoints

import { css } from 'styled-components';

// Devices breakpoints
export const devices = {
  desktop: 1000,
  tablet: 640,
  phone: 420
};

const media = Object.keys(devices).reduce((acc, label) => {
  acc[label] = (strings, ...interpolations) => css`
    @media (max-width: ${devices[label] / 16}em) {
      ${css(strings, ...interpolations)};
    }
  `;
  return acc;
}, {});

export default media;
