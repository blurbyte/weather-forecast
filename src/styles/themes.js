export const dayTheme = {
  primary: '#222222',
  background: '#71d7ed'
};

export const nightTheme = {
  primary: '#ffffff',
  background: '#3033ba'
};
