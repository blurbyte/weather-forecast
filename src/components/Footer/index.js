import React from 'react';

import Text from '../Text';
import Content from './Content';
import Layout from './Layout';

const Footer = () => (
  <Layout>
    <Content>
      <Text dimmed>Data from OpenWeatherMap</Text>
    </Content>
  </Layout>
);

export default Footer;
