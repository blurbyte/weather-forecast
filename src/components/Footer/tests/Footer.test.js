import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

import { dayTheme } from '../../../styles/themes';
import Footer from '../index';

test('renders correctly', () => {
  const component = renderer.create(
    <ThemeProvider theme={dayTheme}>
      <Footer />
    </ThemeProvider>
  );
  expect(component.toJSON()).toMatchSnapshot();
});
