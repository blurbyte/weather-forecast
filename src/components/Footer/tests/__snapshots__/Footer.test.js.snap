// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<footer
  className="sc-bxivhb dECaob"
>
  <div
    className="sc-htpNat kqsjaR sc-bwzfXH ezMGVG"
  >
    <p
      className="sc-bdVaJa bVJipy"
    >
      Data from OpenWeatherMap
    </p>
  </div>
</footer>
`;
